### PR TITLE
DataType and DataBuffer

### PR DESCRIPTION
**Pull request by Niupple**
- Add `DataBuffer` class into the project, and refactored frontend code relating to construction of `Constant` op.
- changed a logging level in `e2etest.py` script from `error` to `info` (which, really seems an info to me)

### CODE REVIEW

File: src/nnfusion/frontend/tensorflow_import/ops/const.cpp [hunk 1]
```diff
@@ -124,6 +125,134 @@ namespace nnfusion
...
+                    }
+                    n_elements *= shape.dim(i).size();
+                }
+
+                // If tensor_content_size is zero, we'll have to take the values from
+                // int_val, float_val, etc.
+                if (tensor_content_size == 0)
+                {
+                    values->resize(n_elements);
+                    for (size_t i = 0; i < n_elements; i++)
```
Comment (Cjkkkk) [inline src/nnfusion/frontend/tensorflow_import/ops/const.cpp:171]: why not just std::memcpy here?

Comment (wenxcs) [reply src/nnfusion/frontend/tensorflow_import/ops/const.cpp:171]: Data from tensorflow pb can only be accessed by its special interface.

File: thirdparty/ngraph/src/nnfusion/common/type/type_protocal.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#include <map>
```
Comment (wenxcs) [inline thirdparty/ngraph/src/nnfusion/common/type/type_protocal.cpp:1]: Add MIT license as other files

**Review by wenxcs — CHANGES_REQUESTED** [review]
Summary:
1. Add license for half to thirdpartynotice.txt file;
2. Add unit test for databuffer;
3. Add missing MIT license for new added files;
4. Change vector<> interface or add databuffer interface for [profiler](https://github.com/microsoft/nnfusion/tree/master/src/nnfusion/engine/profiler);

File: src/nnfusion/frontend/tensorflow_import/ops/const.cpp (hunk 1 above)
Comment (Cjkkkk) [reply src/nnfusion/frontend/tensorflow_import/ops/const.cpp:171]: `.data()` method returns underlying array pointer. see https://github.com/protocolbuffers/protobuf/blob/c981d39d1e3d033c4f00dd48d16e8bdce55a3861/src/google/protobuf/repeated_field.h#L242

Comment (Niupple) [reply src/nnfusion/frontend/tensorflow_import/ops/const.cpp:171]: Thanks for the information, I will look into it.

**Review by wenxcs — APPROVED** [review]

**Pull request merged**